rust/processor/migrations/2024-05-22-200847_add_v1_migration_views/up.sql

```diff
@@ -0,0 +1,327 @@
+        ctov.property_version_v1 AS property_version,
+        ctov.owner_address,
+        ccv.creator_address,
+        ccv.collection_name,
+        ctdv.token_name AS "name",
+        ctov.amount,
+        ctov.token_properties_mutated_v1 AS token_properties,
+        ctov.last_transaction_version,
+        ctov.inserted_at,
+        ctdv.collection_id AS collection_data_id_hash,
+        ctov.table_type_v1 AS table_type,
+        ctov.last_transaction_timestamp
+    FROM current_token_ownerships_v2 ctov
+    JOIN current_token_datas_v2 ctdv
+    ON ctov.token_data_id = ctdv.token_data_id 
+    JOIN current_collections_v2 ccv
+    ON ctdv.collection_id = ccv.collection_id 
+    WHERE ctov.token_standard = 'v1' 
+;
+-- CREATE INDEX lm1_curr_to_collection_hash_owner_index ON legacy_migrations_v1.current_token_ownerships USING btree (collection_data_id_hash, owner_address)
+-- CREATE INDEX lm1_curr_to_crea_cn_name_index ON legacy_migrations_v1.current_token_ownerships USING btree (creator_address, collection_name, name)
+-- CREATE INDEX lm1_curr_to_oa_tt_am_ltv_index ON legacy_migrations_v1.current_token_ownerships USING btree (owner_address, table_type, amount, last_transaction_version DESC)
+-- CREATE INDEX lm1_curr_to_oa_tt_ltv_index ON legacy_migrations_v1.current_token_ownerships USING btree (owner_address, table_type, last_transaction_version DESC)
+
+-- replace `tokens` with `current_token_datas_v2`
+CREATE OR REPLACE VIEW legacy_migration_v1.tokens AS
+    SELECT
+        tdv.token_data_id AS token_data_id_hash,
+        tdv.largest_property_version_v1 AS property_version,
+        tdv.transaction_version,
+        ccv.creator_address,
+        ccv.collection_name,
+        tdv.token_name AS "name",
+        tdv.token_properties,
+        tdv.inserted_at,
+        tdv.collection_id AS collection_data_id_hash,
+        tdv.transaction_timestamp
+    FROM token_datas_v2 tdv
+    JOIN current_token_datas_v2 ctdv
+    ON tdv.token_data_id = ctdv.token_data_id 
+    JOIN current_collections_v2 ccv
+    ON ctdv.collection_id = ccv.collection_id 
+    WHERE tdv.token_standard = 'v1'
+;
+-- CREATE INDEX lm1_token_crea_cn_name_index ON legacy_migration_v1.tokens USING btree (creator_address, collection_name, name)
+
+-- replace `token_datas` with `token_datas_v2`
+CREATE OR REPLACE VIEW legacy_migration_v1.token_datas AS
+    SELECT
+        token_data_id AS token_data_id_hash,
+        tdv.transaction_version,
+        creator_address,
+        collection_name,
+        token_name AS "name",
+        maximum,
+        supply,
+        largest_property_version_v1 AS largest_property_version,
+        token_uri AS metadata_uri,
+        -- Join on new royalty table
+        payee_address,
+        royalty_points_numerator,
+        royalty_points_denominator,
+        TRUE AS maximum_mutable,
+        TRUE AS uri_mutable,
+        TRUE AS description_mutable,
+        TRUE AS properties_mutable,
+        TRUE AS royalty_mutable,
```

Comment:
This is not always TRUE

```
SELECT 
maximum_mutable,
uri_mutable,
description_mutable,
properties_mutable,
royalty_mutable,
COUNT(1)
FROM current_token_datas
GROUP BY 1,2,3,4,5
ORDER BY 1,2,3,4,5
```

Reply:
@bowenyang007 validated that most of these are true, so it's fine to hardcode it to avoid having to migrate the v1 logic over to the v2 processor

Reply:
> these are supposed to be used by marketplaces to indicate which fields are non modifiable. however basically everyone just used one minting contract that sets all of these to true so majority of these ended up being true. and nobody seems to care about these.
my thought is that we can remove them and nothing would happen since I haven’t seen any UI that exposes these. If someone ends up wanting them we could create a new table altogether (similar to royalty and other metadata that aren’t critical).

Field is not nullable, previously these could have different bool values

Reply:
So 2 options here: 
1. make these nullable and set to null. this changes the schema and could cause some compatibility issues
2. keep as is. 

@ying-w do you prefer one or the other?